alpharaw/utils/ms_path_utils.py

```diff
@@ -14,10 +14,23 @@
 ]
 
 
-def get_raw_name(ms_file: str, special_ms_exts: list = _special_ms_exts):
+def get_raw_name(ms_file: str) -> str:
+    """
+    Get raw name str from the ms file path.
+
+    Parameters
+    ----------
+    ms_file : str
+        The ms file path.
+
+    Returns
+    -------
+    str
+        The raw name without extension.
+    """
     raw_name = os.path.basename(ms_file)
     lower_name = raw_name.lower()
-    for _ext in special_ms_exts:
+    for _ext in _special_ms_exts:
         if lower_name.endswith(_ext.lower()):
             raw_name = raw_name[: -len(_ext)]
             break
```

Comment:
is the logic in l. 24-25 required? as far as I get it, this method removes all extensions, not matter if they are defined in `_special_ms_exts` or not?

Reply:
This is required as os.path.splitext cannot recognize `raw_name.d.hdf` this kind of extension formats...

Reply:
alright, but still l.24-25 remove all extensions .. is this desired? or should we limit this to `[.d, .raw, ..]`

Reply:
no, for instance, alphatims save the bruker file as `raw_name.d.hdf`. And as HDF is a general file format, so if we use `*.hdf`, then we don't know if this is a MS data or not. So `*.raw.hdf` or `*.wiff.hdf` is preferred.

Reply:
this is a good example where a unit test would help to gather and implicitly document all the usecases and the desired behaviour of this method :-)

Reply:
That's right